### ISSUE TITLE
[Bug]: When casting varchar to big int, core dumped.

### ISSUE THREAD

**Issue by chrysanthemum-boy** [infiniflow/infinity#612]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

a6ca6a5

### Actual behavior

When deleting table with various type of columns, core dumped.
```
[critical] No implementation to cast from column vector Varchar to big int@src/function/cast/varchar_cast.cppm:238
terminate called after throwing an instance of 'infinity::UnrecoverableException@infinity_exception'
  what():  No implementation to cast from column vector Varchar to big int@src/function/cast/varchar_cast.cppm:238
Aborted (core dumped)
```

### Steps to reproduce

```Markdown
Function test case code can be found in python/test/test_delete.py `test_delete_table_no_rows_met_condition``test_delete_table_all_row_met_the_condition`
Test examples are in python/test/common_values.py
```

**Comment by JinHai-CN** [comment]
Fixed by #609

**Comment by chrysanthemum-boy** [comment]
It looks fixed.
